### ISSUE TITLE
`update` re-updates certain apps every time

### ISSUE THREAD

**Issue by wohfab** [mas-cli/mas#456]
- `mas version`: 9.5
- macOS version: 12.2.1

- [X] `brew install mas` (homebrew-core)

----

Running `mas upgrade`, says, I habe three outdated applications:

```
Honey (14.4.1), Prime Video (9.5), GoodNotes (5.8.6)
```

Software gets upgraded and does not throw any errors. Running `mas upgrade` again, I get the same applications back as "outdated". No matter how often I run the upgrade process, it always says, there are updates.

**Comment by air720boarder** [comment]
This bug is very annoying, but I suspect it's largely with apps that have an iOS counterpart. I have the same issue now with the following apps on my M1 MBP:

1472777122 Honey (14.4.1 -> 14.14.1)
1591303229 Vinegar (2.3.2 -> 2.3.4)
1365531024 1Blocker (4.1.4 -> 5.2.2)

**Comment by coroblac** [comment]
yes, here the same, it would be nice if somebody find a solution for this …

**Comment by StrangeRanger** [comment]
This is similar to my issues. But the thing is, that the application that is outdated, is the current version according to the App Store. I noticed that the app version reported by mas, corresponds the the iOS version of the application.

**Comment by nicerloop** [comment]
For 1Blocker, see https://github.com/mas-cli/mas/issues/468

`{
  "platform": "osx",
  "version": "4.1.6"
}
{
  "platform": "ios",
  "version": "5.2.4"
}`

**Comment by appils** [comment]
okay guys, i'm done with that … i have uninstall mas … inform me please, if it works someday … *mic drop*

**Comment by d7an** [comment]
Adding 3 more that keeps being identified as "outdated"
1548355081 Octane X (10000308.0.2 -> 10.0.3.8.1)
https://apps.apple.com/us/app/octane-x-classic/id1548355081?mt=12

1324203419 Ground News (3.0.1 -> 4.7.0)
https://apps.apple.com/us/app/ground-news/id1324203419

1249901692 ScanScan (3.3.0 -> 3.3.7)
https://apps.apple.com/us/app/scanscan-%E7%99%BD%E6%8F%8F-ocr-text-scanner/id1249901692

**Comment by patricks** [comment]
Another app that re-updates every time

1589391989 Mapper (1.0 -> 2.1)
https://apps.apple.com/at/app/mapper-for-safari/id1589391989?l=en

**Comment by muescha** [comment]
same here:

```shell
$ mas outdated
6444667067 Hyperduck (1.0.4 -> 1.0.5)
640199958 Developer (10.2 -> 10.2.1)

$ mas upgrade
Upgrading 2 outdated applications:
Developer (10.2), Hyperduck (1.0.4)
==> Downloading Apple Developer
==> Installed Apple Developer
==> Downloading Hyperduck
==> Installed Hyperduck

$ mas outdated
6444667067 Hyperduck (1.0.4 -> 1.0.5)
640199958 Developer (10.2 -> 10.2.1)
```

Note: the Apps in App Store are with this versions. I don't know where `mas` get the new version numbers...
HyperDuck -> 1.0.4
Developer -> 10.2

**Comment by rinchen** [comment]
same
`~ mas outdated                                                                                                                            1 [12:02:43]
1503642685 Passwarden (3.1.0 -> 3.1.8)
1475622766 SmartDNS (1.12.3 -> 1.12.6)
1553438508 DNS Firewall (1.0.6 -> 1.0.7)
694633015 VPN Unlimited® (8.7.0 -> 9.1.14)`

**Comment by rgoldberg** [comment]
Closing as a duplicate of #491. PR #496 fixed the issue. It will be released in the upcoming version 1.8.7, which is currently in beta.

If you think this issue is different than #491, please let us know.

**Comment by rgoldberg** [comment]
Closing as a duplicate of #491. PR #496 fixed the issue. It will be released in the upcoming version 1.8.7, which is currently in beta.

If you think this issue is different than #491, please let us know.
